base/client.jl

```diff
@@ -231,9 +231,18 @@ let reqarg = Set(UTF8String["--home",          "-H",
                             "--bind-to"])
     global process_options
     function process_options(opts::JLOptions, args::Vector{UTF8String})
-        if !isempty(args) && (arg = first(args); !isempty(arg) && arg[1] == '-' && in(arg, reqarg))
-            println(STDERR, "julia: option `$arg` is missing an argument")
-            exit(1)
+        if !isempty(args)
+            arg = first(args)
+            if !isempty(arg) && arg[1] == '-' && in(arg, reqarg)
+                println(STDERR, "julia: option `$arg` is missing an argument")
+                exit(1)
+            end
+            idxs = find(x -> x == "--", args)
```

Comment:
No reason. @vtjnash can you point to a source that explains this in more detail or gives an example?  I've never seen what you have pointed out in the wild.